pkg/limayaml/defaults.go

```diff
@@ -334,20 +334,28 @@ func FillDefault(y, d, o *LimaYAML, filePath string) {
 	iface := make(map[string]int)
 	for _, nw := range append(append(d.Networks, y.Networks...), o.Networks...) {
 		if i, ok := iface[nw.Interface]; ok {
-			if nw.VNL != "" {
-				networks[i].VNL = nw.VNL
-				networks[i].SwitchPort = nw.SwitchPort
+			if nw.VNLDeprecated != "" {
+				networks[i].VNLDeprecated = nw.VNLDeprecated
+				networks[i].SwitchPortDeprecated = nw.SwitchPortDeprecated
+				networks[i].Socket = ""
+				networks[i].Lima = ""
+			}
+			if nw.Socket != "" {
+				networks[i].Socket = nw.Socket
+				networks[i].VNLDeprecated = ""
+				networks[i].SwitchPortDeprecated = nw.SwitchPortDeprecated
 				networks[i].Lima = ""
 			}
 			if nw.Lima != "" {
-				if nw.VNL != "" {
+				if nw.VNLDeprecated != "" {
 					// We can't return an error, so just log it, and prefer `lima` over `vnl`
 					logrus.Errorf("Network %q has both vnl=%q and lima=%q fields; ignoring vnl",
-						nw.Interface, nw.VNL, nw.Lima)
+						nw.Interface, nw.VNLDeprecated, nw.Lima)
 				}
```

Comment:
Should there be another test here to verify that `nw.Lima` and `nw.Socket` are not both set at the same time?

Although I don't know why we have this test here at all, as it should already have been covered by `limayaml.validateNetwork`.

Reply:
Fixed